app/builders/messages/facebook/message_builder.rb

```diff
@@ -43,7 +43,8 @@ def build_contact
     return if contact.present?
 
     @contact = Contact.create!(contact_params.except(:remote_avatar_url))
-    @contact_inbox = ContactInbox.find_or_create_by!(contact: contact, inbox: @inbox, source_id: @sender_id)
+    @contact_inbox = ContactInbox.find_or_create_by!(inbox: @inbox, source_id: @sender_id)
```

Comment:
@tejaswinichile, could you add more details on this change ? Do we know how we end up with contact inboxes with the same source_id but a different contact associated with it?

Reply:
@sojan-official I am not so sure about this, because we enter this block when contact is not present.
But I was memorizing contact [here](https://github.com/chatwoot/chatwoot/pull/5507/files#diff-0d098b3b099261200b5b9c56cb08c4eaebfea4413d8200934c94cd286f7e4bf9R39) so I think that might be the case and also didn't add the attr_accessor for the same method.

I think some memory saved from the different contact might be affecting the same block. But with this fix, we are trying to avoid the situation.

Reply:
@tejaswinichile let's try to clearly understand what's happening and how to replicate this case.

Reply:
@sojan-official One of my guesses is what if an Instagram user or FB user changes their name later and then tries to send the message? I will test this out in our staging and add the investigation here.